service/biz/enrollz_biz_test.go

```diff
@@ -0,0 +1,281 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package biz
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"google.golang.org/protobuf/encoding/prototext"
+	"google.golang.org/protobuf/testing/protocmp"
+
+	cpb "github.com/openconfig/attestz/proto/common_definitions"
+	epb "github.com/openconfig/attestz/proto/tpm_enrollz"
+)
+
+type stubEnrollzInfraDeps struct {
+	SwitchOwnerCaClient
+	EnrollzDeviceClient
+
+	// Request params that would be captured in stubbed deps' function calls.
+	cardIdIssueIakReq    *cpb.ControlCardVendorId
+	cardIdIssueIDevIdReq *cpb.ControlCardVendorId
+	iakPubPemReq         string
+	iDevIdPubPemReq      string
+	getIakCertReq        *epb.GetIakCertRequest
+	rotateOIakCertReq    *epb.RotateOIakCertRequest
+
+	// Stubbed responses to simulate behavior of deps without implementing them.
+	oIakCertResp       string
+	oIDevIdCertPemResp string
+	getIakCertResp     *epb.GetIakCertResponse
+	rotateOIakCertResp *epb.RotateOIakCertResponse
+
+	// If we need to simulate an error response from any of the deps, then set
+	// the dep's response to nil and populate this error field.
+	errorResp error
+}
+
+func (s *stubEnrollzInfraDeps) IssueOwnerIakCert(cardId *cpb.ControlCardVendorId, iakPubPem string) (string, error) {
+	// Validate that no stub (captured) request params were set prior to execution.
+	if s.cardIdIssueIakReq != nil {
+		return "", fmt.Errorf("IssueOwnerIakCert unexpected req cardId %s", prototext.Format(s.cardIdIssueIakReq))
+	}
+	s.cardIdIssueIakReq = cardId
+
+	if s.iakPubPemReq != "" {
+		return "", fmt.Errorf("IssueOwnerIakCert unexpected req IAK pub key PEM %s", s.iakPubPemReq)
+	}
+	s.iakPubPemReq = iakPubPem
+
+	// If the response is set, then return it, otherwise return error.
+	if s.oIakCertResp == "" {
+		return "", s.errorResp
+	} else {
```

Comment:
nit: Can remove this else (and many subsequent ones further in the code)

Reply:
Sounds good, will do!